src/__tests__/utils/before-capture-utils.test.ts

```diff
@@ -0,0 +1,57 @@
+import { sampleByDistinctId, sampleByEvent, sampleBySessionId } from '../../utils/before-capture.utils'
+import { CaptureResult } from '../../types'
+import { isNull } from '../../utils/type-utils'
+
+function expectRoughlyFiftyPercent(emittedEvents: any[]) {
+    expect(emittedEvents.length).toBeGreaterThanOrEqual(40)
+    expect(emittedEvents.length).toBeLessThanOrEqual(60)
+}
+
+describe('before capture utils', () => {
+    it('can sample by event name', () => {
+        const sampleFn = sampleByEvent(['$autocapture'], 50)
+        const results = []
+        Array.from({ length: 100 }).forEach(() => {
+            const captureResult = { event: '$autocapture' } as unknown as CaptureResult
+            results.push(sampleFn(captureResult))
+        })
+        const emittedEvents = results.filter((r) => !isNull(r))
+        expectRoughlyFiftyPercent(emittedEvents)
```

Comment:
Can we mock Math.random for this so that it won't fail (very) occasionally?
```
import prand from 'pure-rand';
const mockRandom = (seed=42) => {
    const rng = prand.xoroshiro128plus(seed);
    return () => prand.unsafeUniformIntDistribution(0, 1_000_000, rng) / 1_000_000
}
```

Reply:
Ha your new approach also works!